### PR TITLE
7 sync with icloud

### PR DESCRIPTION
**Pull request by namolnad**
Hey @mhanlon I spent a bunch of time over the past week working on understanding how to interact with iCloud and how to transfer over to that from Core Data. I think I ended in a pretty good place with minimal bugs. Let me know what you think and if all looks good to merge into master! 

Thanks,
Dan

PS though everything was working perfectly not long ago, I did just reorganize code and files to make it easier to read through/understand. Though there would only be a very, very slight possibility that I broke something in that process, I thought I should warn that I haven't tested this most up-to-date version quite yet.

### CODE REVIEW

File: blanknote/blanknote/AppDelegate.m [hunk 1]
```diff
@@ -10,9 +10,14 @@
 #import "DetailViewController.h"
 #import "MasterViewController.h"
 #import "CoreDataEngine.h"
+#import "BlanknoteDefinitions.h"
+#import "FormattingEngine.h"
 
 @interface AppDelegate () <UISplitViewControllerDelegate>
 
+@property (nonatomic, strong) NSUserDefaults *prefs;
```
Comment (mhanlon) [inline blanknote/blanknote/AppDelegate.m]: No need to make this a `strong` reference, as it's a shared instance, and will be around for the life of the app, anyway.

File: blanknote/blanknote/AppDelegate.m [hunk 2]
```diff
@@ -10,9 +10,14 @@
...
+@property (nonatomic, strong) UINavigationController *navigationController;
```
Comment (mhanlon) [inline blanknote/blanknote/AppDelegate.m]: Probably the same comment (no need for the `strong` reference) here, though I don't see anything ever assigned to this property...

File: blanknote/blanknote/DetailViewController.m [hunk 3]
```diff
@@ -135,36 +174,148 @@ - (void)setTextViewToDetectData:(BOOL)flag
...
+                            object:nil];
+    
+    [notifcationCenter addObserver:self
+                          selector:@selector(keyboardWillBeHidden:)
+                              name:UIKeyboardWillHideNotification
+                            object:nil];
+}
+
+- (void) deregisterForNotifcations
+{
```
Comment (mhanlon) [inline blanknote/blanknote/DetailViewController.m]: The handy thing here is we can just call `[notifcationCenter removeObserver:self]` and that will take us off all notifications we registered for...

**Comment by mhanlon** [comment]
This is looking good. There's that minor issue with the iCloud setting, but otherwise regular operation looks good.

Sharing a note into the app isn't saving it in the iCloud store (it's saving it to the `group.banknote` shared store), which is more of an issue, but the only one I'd find that I'd want to fix before I ship the app.

**Comment by namolnad** [comment]
Thanks Matt,

I've fixed most of it, with the exception of the sharing issue. For some
reason it looks as though the nsuserdefaults is showing the wrong settings
for the bool in question. I've tried to create a suite like so, but haven't
figured it out quite yet. Seem to be getting crashes since introducing that
change...

    NSUserDefaults *prefs = [[NSUserDefaults alloc] initWithSuiteName:
@"group.blanknote"];

But

**Comment by namolnad** [comment]
Was able to fix that crash, which was actually coming from the entitlements for the share extension. (the default container added ".blanknoteSharing" to the end of the entitled containers...

There is now no crash, however, it seems that the share extension is creating it's own persistent store, separate from the one created by iCloud. I've been researching it and trying a few different things and haven't been able to find an answer. I've seen a few people say that iCloud and share extensions haven't worked well together in the past, but I'm not sure why that would be the case still. Any thoughts?

**Comment by namolnad** [comment]
Here is what I'm seeing over here:

[3] (null)  @"PFUbiquitySetupOriginalStoreURLAbsoluteStringKey" : @"file:///Users/Pantheon/Library/Developer/CoreSimulator/Devices/D0B9B760-7188-4407-9D5E-B6814C1A9B80/data/Library/Mobile%20Documents/iCloud~com~namolnad~blanknote/iCloudBlanknoteStore.sqlite"  
[8] (null)  @"PFUbiquitySetupContainerIdentifierKey" : @"E80DEBDA-0D23-4159-86B2-792DE994B0DC"  
[11]    (null)  @"NSStoreUUID" : @"7CA595D7-42CC-472C-9DED-F1C55642603B"    

[3] (null)  @"PFUbiquitySetupOriginalStoreURLAbsoluteStringKey" : @"file:///Users/Pantheon/Library/Developer/CoreSimulator/Devices/D0B9B760-7188-4407-9D5E-B6814C1A9B80/data/Library/Mobile%20Documents/iCloud~com~namolnad~blanknote/iCloudBlanknoteStore.sqlite"  
[8] (null)  @"PFUbiquitySetupContainerIdentifierKey" : @"A82F62D8-CDAB-440E-9F2E-9B1116DBDE5A"  
[11]    (null)  @"NSStoreUUID" : @"C3B46822-570F-43A9-9199-FEC43457565A"

**Comment by mhanlon** [comment]
Good strategy... a few spurious bugs to work out, but nothing to stop this from completing the checkpoint!

**Pull request merged**